unit-test/fm_child_tests.c

```diff
@@ -3512,6 +3515,61 @@ void Test_FM_ChildLoop_CountSemTakeSuccessDefault(void)
  * ChildSleepStat Tests
  * ***************/
 
+void Test_FM_ChildSleepStat_getSizeTimeModeFalse(void)
+{
+    // Arrange
+    const char *dummy_filename          = "dummy_filename";
+    FM_DirListEntry_t dummy_DirListData = {.EntrySize=1, .ModifyTime=1, .Mode=1};
+    int32 dummy_FilesTillSleep          = 1;
+    uint8 dummy_getSizeTimeMode         = 0;
+
+    // Act
+    FM_ChildSleepStat(dummy_filename, &dummy_DirListData, &dummy_FilesTillSleep, dummy_getSizeTimeMode);
+    
+    // Assert
+    UtAssert_INT32_EQ(dummy_DirListData.EntrySize, 0);
+    UtAssert_INT32_EQ(dummy_DirListData.ModifyTime, 0);
+    UtAssert_INT32_EQ(dummy_DirListData.Mode, 0);
+}
+
+void Test_FM_ChildSleepStat_FilesTillSleepPositive(void)
+{
+    // Arrange
+    const char *dummy_filename          = "dummy_filename";
+    FM_DirListEntry_t dummy_DirListData = {.EntrySize=1, .ModifyTime=1, .Mode=1};
+    int32 dummy_FilesTillSleep          = FM_CHILD_STAT_SLEEP_FILECOUNT + 1;
+    uint8 dummy_getSizeTimeMode         = 1;
+    int32 dummy_FilesTillSleep_before   = dummy_FilesTillSleep;
+
+    // Act
+    FM_ChildSleepStat(dummy_filename, &dummy_DirListData, &dummy_FilesTillSleep, dummy_getSizeTimeMode);
+    
+    // Assert
+    UtAssert_STUB_COUNT(FM_ChildSizeTimeMode, 1);
```

Comment:
@skliper Do you know if I'm testing the call to `FM_ChildSizeTimeMode()` incorrectly here or on line 3567 of the proposed changes? Both of these instances fail when I run them, however, I believe they should pass given that these 2 tests reach the call below https://github.com/nasa/FM/blob/51707f20350e4ea909842acb3597fff8da3ab18e/fsw/src/fm_child.c#L1633

Reply:
Ah, you won't get a stub count for functions that aren't stubbed.  Since FM_ChildSizeTimeMode is implemented in the unit under test the real code will be called.